src/wallet/scriptpubkeyman.cpp

```diff
@@ -271,6 +271,8 @@ bool LegacyScriptPubKeyMan::GetReservedDestination(const OutputType type, bool i
     if (!ReserveKeyFromKeyPool(index, keypool, internal)) {
         return false;
     }
+    LearnRelatedScripts(keypool.vchPubKey, type);
+    address = GetDestinationForKey(keypool.vchPubKey, type);
```

Comment:
In commit "Key pool: Move LearnRelated and GetDestination calls" (35c37b59c5bb918c15aca3092b6506ef3974e50c)

Maybe point out in the commit description that addresses are determined by LegacyScriptPubKeyMan::GetReservedDestination 
instead of ReserveDestination::GetReservedDestination because other key manager implementations might construct them in other ways.

Reply:
Done